src/main/java/org/elasticsearch/cluster/ClusterState.java

```diff
@@ -600,6 +647,13 @@ public ClusterState build() {
             return os.bytes().toBytes();
         }
 
+        public static byte[] toDiffBytes(ClusterState previousState, ClusterState state) throws IOException {
```

Comment:
not sure where this is used but would it make more sense to just offer a method to write this into a stream?